### PR TITLE
Feature/m2 1085 create update applet integration

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/modules/Builder/features/AboutApplet/AboutApplet.tsx [hunk 1]
```diff
@@ -33,7 +34,22 @@ const commonUploaderProps = {
...
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation();
+  const { result: appletData } = applet.useAppletData() ?? {};
+  const isNewApplet = useCheckIfNewApplet();
+  const defaults = isNewApplet
+    ? defaultValues
+    : ({
+        displayName: appletData?.displayName ?? '',
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/AboutApplet/AboutApplet.tsx:46]: could you pls try to distract values like this:
const {displayName = '', description = '', ... } = appletData;

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { BuilderLayers, useCheckIfNewApplet } from 'shared/hooks';
+import { builderSessionStorage, Path } from 'shared/utils';
+import { applet } from 'shared/state';
+
+import {
+  appletActivitiesMocked,
+  appletActivityFlowsMocked,
+  appletDataMocked,
+  appletPasswordMocked,
+} from './mock';
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts:13]: do we need mocked values anymore? maybe it's good idea to get rid of them?

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.tsx [hunk 3]
```diff
@@ -1,14 +1,58 @@
...
+      result = await dispatch(createApplet(body));
+    }
+    if (!isNewApplet && appletId) {
+      result = await dispatch(updateApplet({ appletId, body }));
+    }
+    if (!result) return;
+
+    clearStorage();
+    // eslint-disable-next-line
+    // @ts-ignore
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/SaveAndPublish/SaveAndPublish.tsx:43]: pls avoid using ts-ignore
could you fix pls?

File: src/shared/state/Applet/Applet.reducer.ts [hunk 4]
```diff
@@ -13,4 +13,12 @@ export const extraReducers = (builder: ActionReducerMapBuilder<AppletSchema>): v
   createAppletPendingData({ builder, thunk: getApplet, key: 'applet' });
   createAppletFulfilledData({ builder, thunk: getApplet, key: 'applet' });
   createAppletRejectedData({ builder, thunk: getApplet, key: 'applet' });
+
+  createAppletPendingData({ builder, thunk: createApplet, key: 'applet' });
```
Comment (IhorNesterenko99) [inline src/shared/state/Applet/Applet.reducer.ts:17]: I thin it's better to use common: 
appletPendingData, appletFullfilledData, appletRejectedData

File: src/shared/utils/sessionStorage.ts [hunk 5]
```diff
@@ -1,13 +1,13 @@
 import { BuilderLayers } from 'shared/hooks';
 
-const getItemName = (layer: BuilderLayers) => `builder_${layer}_data`;
-const setSpecificItem = (layer: BuilderLayers, data: Record<string, unknown>) => {
+const getItemName = (item: BuilderLayers) => `builder_${item}_data`;
+const setSpecificItem = (item: BuilderLayers, data: /*boolean | */ unknown) => {
```
Comment (IhorNesterenko99) [inline src/shared/utils/sessionStorage.ts:4]: do we need this commented code?
/*boolean | */

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts (hunk 2 above)
Comment (alex-i-e) [reply src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts:13]: yes, we need the mocked data for a while till the BE will fix the issue with activities https://mindlogger.atlassian.net/browse/M2-1139 and will add a separate API to retrieve items https://mindlogger.atlassian.net/browse/M2-1125?

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Pull request merged**